libsolidity/formal/BMC.cpp

```diff
@@ -47,8 +47,11 @@ BMC::BMC(
 	CharStreamProvider const& _charStreamProvider
 ):
 	SMTEncoder(_context, _settings, _errorReporter, _unsupportedErrorReporter, _charStreamProvider),
-	m_interface(make_unique<smtutil::SMTPortfolio>(_smtlib2Responses, _smtCallback, _settings.solvers, _settings.timeout))
+	m_interface(make_unique<smtutil::SMTPortfolio>(
+		_smtlib2Responses, _smtCallback, _settings.solvers, _settings.timeout, _settings.printQuery
+	))
 {
+	solAssert(!_settings.printQuery || _settings.solvers.smtlib2, "Must use SMTLib2 solver to dump queries");
```

Comment:
Is this reachable by the user by just passing these flags?

Reply:
This is checking that `smtlib2` must be enabled, but I think it should check that **only** `smtlib2` is enabled.

Reply:
@leonardoalt done